### ISSUE TITLE
Suppress chain related logs

### ISSUE THREAD

**Issue by gege251** [mlabs-haskell/plutip#48]
cardano-node and chain-index runs in a separate process and has access to stdout, printing out logs to the screen, polluting the test results. Ideally, we would use a `verbose` flag, or log levels to control this.

**Comment by zmrocze** [comment]
This messy logging output consists of interleaved:

1. cluster-logs, these are logged both to a file and to std outputs [as seen here](https://github.com/mlabs-haskell/plutip/blob/1c506941fb2572af50e892188f6116e6509fb253/src/Test/Plutip/Internal/LocalCluster.hs#L150).

2. cardano-node log from before logging is initialized. [cardano-node/Run.hs](https://github.com/input-output-hk/cardano-node/blob/9c991b0c610b5a0f2b58e0c381e9ac7c4b11ac78/cardano-node/src/Cardano/Node/Run.hs#L122)

In 1) we probably simply want to remove `LogToStdStreams` from the list of logging outputs.

Log from 2) is a result of a "putStrLn" call that prints out a huge `NodeConfiguration` record. As long as the child process uses the same stdout as the test process, this will get printed. I haven't found yet the place where a new process for cardano-node is forked. If it is inside plutip code, then we could swap node's stdout for some other file handle.

**Comment by zmrocze** [comment]
Okey right, there is also chain-index as a third source of logging.

Let me show you what I did so far. I pr'ed a hack [here](https://github.com/mlabs-haskell/plutip/pull/57) where we:
1. Remove stdout log output passed to cluster, so that the cluster now only logs to a log file.
2. Wrap call to cardano-wallet's `withCluster` in a function that swaps out stdout file handle for provided handle to a file "node_configuration.log" in temp directory.

One can inspect "node_configuration.log" file by setting environment variable "NO_CLEANUP" to leave tmp dir unremoved.

Though with this fix the function `withPlutusInterface` is not thread-safe, it swaps stdout back and forth and other threads can observe various stdout handles. This happens already as the chain-index runs in second thread. It does some printing and sometimes it ends up in the "node_configuration.log" file. So 2. is indeed a hack.

As for the chain-index: Should I pr plutus-apps and add verbose flag to chain-index? This would remove the third source of unwanted logs. I've tried quickly running chain-index with `forkProcess` instead of `async`, but this ofc introduced some new problems.
